lib/cartopy/tests/test_shapereader.py

```diff
@@ -59,18 +62,20 @@ def test_geometry(self):
         assert len(geometries) == len(self.reader)
 
         # Choose a nice small lake
-        lake = geometries[14]
+        lake = geometries[self.lake_index]
         self._assert_geometry(lake)
 
     def test_record(self):
         records = list(self.reader.records())
         assert len(records) == len(self.reader)
 
         # Choose a nice small lake
-        lake_record = records[14]
-        assert lake_record.attributes.get('name') == 'Lago de\rNicaragua'
-        assert sorted(lake_record.attributes.keys()) == \
-            sorted(['admin', 'featurecla', 'scalerank', 'name_alt', 'name'])
+        lake_record = records[self.lake_index]
+        assert lake_record.attributes.get('name') == self.lake_name
+        expected = sorted(['admin', 'featurecla', 'min_label', 'min_zoom',
+                           'name', 'name_alt', 'scalerank'])
+        actual = sorted(lake_record.attributes.keys())
+        assert actual == expected
         lake = lake_record.geometry
         self._assert_geometry(lake)
```

Comment:
Not your fault, and maybe I shouldn't rock the boat, but ...
The whole definition + use of `_assert_geometry` smells a bit, and same for the TestRivers class !

It should really be sufficient to do the magic number testing wholly in `test_geometry` (no separate `_assert_geometry` method).
Then in `test_record`, instead of calling `_assert_geometry` again I think we should just have something like `assertEqual(self.test_record.geometry, self.test_geometry)`.
For that, we should probably push the initial evaluation of `list(self.reader.geometries())` and 'list(self.reader.records())' up into the setUp call.

Code suggestions here ... : https://github.com/djkirkham/cartopy/pull/2